### DISCUSSION TITLE
A question about roughness usage

### DISCUSSION THREAD

**Discussion by zhang-wenling** [firemodels/fds#13377] [Q&A]
Hi, 
I've been working on some tunnel models that use `HVAC` to achieve 1D-3D combination recently, and I want to show the phenomenon of tunnel wind speed attenuation, I want to use `roughness` to achieve it. But the `Velocity_MASS MEAN` data I got was weird.

![image](https://github.com/user-attachments/assets/60c5f825-f5e9-459e-92fb-05ced9583185)
![image](https://github.com/user-attachments/assets/b0088746-6fe1-490f-a6a2-8ecf26fefe54)

These are two simulated codes：
[1.txt](https://github.com/user-attachments/files/16823836/1.txt)
[2.txt](https://github.com/user-attachments/files/16823844/2.txt)

**Comment by mcgratta** [comment]
Are you setting `TUNNEL_PRECONDITIONER=T` on the `PRES` line? If not, read more about this parameter in the User's Guide.

**Reply by zhang-wenling** [reply]
Thanks for the suggestion, I'm using the fds version of 6.7.4. Can I only use a higher fds version to implement this code, because I found that fds6.7.4 does not have this code, is there any other code in fds6.7.4 instead of `TUNNEL_PRECONDITIONER=T` or is there another way to solve this problem in fds6.7.4. Can I use the GLMAT solver mentioned in the fds user guide to solve the problem to reduce the error error?

**Comment by drjfloyd** [comment]
2.fds
Generated by PyroSim 2023.3.1206

If you are using the FDS installed with the version of Pyrosim you are using, then you are also using an old version of FDS which won't support the feature Kevin mentioned. Make sure to install and run with the current release.

**Reply by zhang-wenling** [reply]
Thanks for the suggestion, I'm using the fds version of 6.7.4. Can I only use a higher fds version to implement this code, because I found that fds6.7.4 does not have this code, is there any other code in fds6.7.4 instead of `TUNNEL_PRECONDITIONER=T` or is there another way to solve this problem in fds6.7.4. Can I use the GLMAT solver mentioned in the fds user guide to solve the problem to reduce the error error?

**Reply by drjfloyd** [reply]
Use the current release, read that version's User's Guide discussion on tunnels (which discussed the preconditioner and other options, and adjust your input accordingly. See if that solves your problem.
